app/controllers/attributes/names_controller.rb

```diff
@@ -1,13 +1,8 @@
 class Attributes::NamesController < AttributesController
   def show
-    local_attributes, remote_attributes = get_attributes_from_params(
-      params.fetch(:attributes),
-      permission_level: :check,
-    )
+    attributes = params.fetch(:attributes)
+    validate_attributes!(attributes, :check)
```

Comment:
Excellent, this makes fr much eater controllers too.

Could even go further with something like:

by putting `attributes = params.fetch(:attributes)` into the Concern, memoising it to an instance variable and then using a before action?

Reply:
Sure